src/net/config.rs

```diff
@@ -19,6 +19,10 @@ pub struct NetworkConfig {
     /// Enable adding loopback addresses to the address book. Should be
     /// enabled during testing and disabled in production.
     pub enable_loopback: bool,
+    /// Manage addresses in the address book automatically. This removes
+    /// them when an address is unreachable and removes the peer when there
+    /// is a dial failure.
+    pub manage_addresses: bool,
```

Comment:
It's mostly about automatically pruning the address book, right? Maybe `prune_addresses`?